tests/integration/ceph_bucket_notification_test.go

```diff
@@ -122,38 +110,187 @@ func (s *ObjectSuite) TestBucketNotificationsInOrder() {
+		assert.Nil(t, err)
+		delete(obc.Labels, bucketNotificationLabelPrefix+notificationName)
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		// check whether existing bucket notification uneffected
+		var notificationPresent bool
+		for i := 0; i < 4; i++ {
+			notificationPresent = helper.BucketClient.CheckBucketNotificationSetonRGW(namespace, storeName, obcName, bucketname, notificationName, s3endpoint)
+			if !notificationPresent {
+				break
+			}
+			time.Sleep(5 * time.Second)
 		}
+		assert.False(t, notificationPresent)
+	})
 
+	t.Run("add new topic, notification to existing ObjectBucketClaim", func(t *testing.T) {
+		newNotificationName := "new-notification"
+		newTopicName := "new-topic"
+		t.Run("create CephBucketTopic: new-topic", func(t *testing.T) {
+			err := helper.TopicClient.CreateTopic(newTopicName, storeName, httpEndpointService)
+			assert.Nil(t, err)
+			created := utils.Retry(12, 2*time.Second, "topic is created", func() bool {
+				return helper.TopicClient.CheckTopic(newTopicName)
+			})
+			assert.True(t, created)
+		})
+		t.Run("create CephBucketNotification: new-notification", func(t *testing.T) {
+			err = helper.NotificationClient.CreateNotification(newNotificationName, newTopicName)
+			assert.Nil(t, err)
+			created := utils.Retry(12, 2*time.Second, "notification is created", func() bool {
+				return helper.NotificationClient.CheckNotificationCR(newNotificationName)
+			})
+			assert.True(t, created)
+		})
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
 		assert.Nil(t, err)
-		logger.Infof("endpoint (%s) Accesskey (%s) secret (%s) region (%s)", s3endpoint, s3AccessKey, s3SecretKey, rgwcontext.ZoneGroup)
+		obc.Labels[bucketNotificationLabelPrefix+newNotificationName] = newNotificationName
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		t.Run("check CephBucketNotification new-notification created for bucket", func(t *testing.T) {
+			// check whether bucket notification added
+			notificationPresent := utils.Retry(12, 2*time.Second, "notification is created for bucket", func() bool {
+				return helper.BucketClient.CheckBucketNotificationSetonRGW(namespace, storeName, obcName, bucketname, newNotificationName, s3endpoint)
+			})
+			assert.True(t, notificationPresent)
+		})
+		t.Run("delete CephBucketNotification: new-notification", func(t *testing.T) {
+			err = helper.NotificationClient.DeleteNotification(newNotificationName, topicName)
+			assert.Nil(t, err)
+		})
+		t.Run("delete CephBucketTopic: new-topic", func(t *testing.T) {
+			err = helper.TopicClient.DeleteTopic(newTopicName, storeName, httpEndpointService)
+			assert.Nil(t, err)
+		})
```

Comment:
I am not quite  sure @yuvalif do the backend notifications are removed when bucket notifications is CR removed. Or this will be removed only actual bucket is remvoed

Reply:
* this specific line is for topic CR deletion, not notification CR deletion
* cascade deletion of notification when the notification CR is deleted is not implemented yet. it is planned to be at [phase2](https://gist.github.com/yuvalif/c4c7aed3dbc956a65399865a77c6a929#priority-2)
* when a bucket is removed, all associated notifications are removed (this is done inside the RGW)

Reply:
Let's add a `t.Run()` placeholder with a TODO comment and `t.Skip()` to know to come back later and verify then.

Reply:
Sorry I am not clear here, from @yuvalif comment topic CR deletion is not result in deleting backend notifications or the notification CR. The issue is even if we delete notification CR with current code the backend notification still exists. This is removed only when bucket is removed. I will add `t.Run() and TODO with t.Skipped()` as part of notification CR deletion tests cases. But not here. Am I missing something?